[Bug] nil pointer exception thrown in `Broadcast` when `BroadcastTxSync` returns nil `ResultBroadcastTx`

In `Broadcast` a nil pointer exception is thrown on L140 when attempting to log errors if the `BroadcastTxSync` call returns a nil `*coretypes.ResultBroadcastTx`

The call where the NPE is encountered:
```go
logger.Error(fmt.Sprintf("received non zero : %d - %s", rpcResponse.Code, rpcResponse.Log))
```

Logs:
```
noble-cctp-relayer start --config .ignore/testnet.yaml
3:22PM INF successfully parsed config file location=.ignore/testnet.yaml
3:22PM INF wss://goerli.infura.io/ws/v3/d33a2875a8e6483da6ce10fe607eba31
3:22PM INF Starting Ethereum listener at block 10015403 looking back 100 blocks
3:22PM INF New historical msg from source domain 0 with tx hash 0x0b4841d3d74a464c54a77a5d9a7eba71397b06c9aaaee13a3e40590be03d346e
3:22PM INF New historical msg from source domain 0 with tx hash 0x4caaf1d3e078ad137044740dd8c6df2e7e0ed1064c3425c07ecdcbb6fe184796
3:22PM INF New historical msg from source domain 0 with tx hash 0xa931a1f24cd60ab8dd8a4048467e8f93ee2aa2610b03ffc43fcb99226747f2d0
3:22PM INF New historical msg from source domain 0 with tx hash 0x80e55624bdf935d49c8991a873e8399640b0178c4b66970bc65932d97e66be06
3:22PM INF New historical msg from source domain 0 with tx hash 0x3a1eef6d2d6a1f928aae585d96ccbe953d6e43f56e0e020e9ae164b6e0244308
3:22PM INF New historical msg from source domain 0 with tx hash 0x20cff2cf746f6ce70d62776626afa7b61646b406db28685c395ac29490dae01f
3:22PM INF New historical msg from source domain 0 with tx hash 0x3c2d3412f940b7f0479849fea6f3d36ec80afd9a007eed094591afedf6ef0584
3:22PM INF New historical msg from source domain 0 with tx hash 0x91a204327367f5be51a1f3199aa455ff70daaa5cea1505724de4b1229efb4436
3:22PM INF New historical msg from source domain 0 with tx hash 0x1acbfd072c5d6c0a24c58d7465da4bab475dbed664722826f6ec47204b5ac7e4
3:22PM INF New historical msg from source domain 0 with tx hash 0x91cedffd4211aef6873fd74cb54d2c799d4408fa569060a7931ef06cc80c900f
3:22PM INF Filtered tx 0x4caaf1d3e078ad137044740dd8c6df2e7e0ed1064c3425c07ecdcbb6fe184796 because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0x3a1eef6d2d6a1f928aae585d96ccbe953d6e43f56e0e020e9ae164b6e0244308 because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0x20cff2cf746f6ce70d62776626afa7b61646b406db28685c395ac29490dae01f because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0x80e55624bdf935d49c8991a873e8399640b0178c4b66970bc65932d97e66be06 because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0x91a204327367f5be51a1f3199aa455ff70daaa5cea1505724de4b1229efb4436 because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0xa931a1f24cd60ab8dd8a4048467e8f93ee2aa2610b03ffc43fcb99226747f2d0 because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0x0b4841d3d74a464c54a77a5d9a7eba71397b06c9aaaee13a3e40590be03d346e because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0x1acbfd072c5d6c0a24c58d7465da4bab475dbed664722826f6ec47204b5ac7e4 because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0x3c2d3412f940b7f0479849fea6f3d36ec80afd9a007eed094591afedf6ef0584 because relaying from 0 to 7 is not enabled
3:22PM INF Filtered tx 0x91cedffd4211aef6873fd74cb54d2c799d4408fa569060a7931ef06cc80c900f because relaying from 0 to 7 is not enabled
3:22PM INF Starting Noble listener at block 3841811
3:23PM INF New stream msg from 0 with tx hash 0x0a05cf1b806f16815bb012783c3b0ed7afd9408e4ca4d0c27b5c8d067f08342d
3:23PM INF Filtered tx 0x0a05cf1b806f16815bb012783c3b0ed7afd9408e4ca4d0c27b5c8d067f08342d because relaying from 0 to 7 is not enabled
3:25PM INF New stream msg from 0 with tx hash 0xbb1d2cd7998f6ed1241fd61198d03076518111a0f3f118eebc2df24049ed6b11
3:25PM INF Filtered tx 0xbb1d2cd7998f6ed1241fd61198d03076518111a0f3f118eebc2df24049ed6b11 because relaying from 0 to 7 is not enabled
3:27PM INF New stream msg from 0 with tx hash 0x6882c80ebe162822e2b1720b76c303131f3d5ab3319381dd09d27d264734093f
3:28PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:28PM INF New stream msg from 0 with tx hash 0xe5533dd511d1f15fca6be861dddc9d8dd174deb2365978062f6438547657507b
3:28PM INF Filtered tx 0xe5533dd511d1f15fca6be861dddc9d8dd174deb2365978062f6438547657507b because relaying from 0 to 7 is not enabled
3:28PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:28PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:28PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:28PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:29PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:29PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:29PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:29PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:29PM INF Attestation found for https://iris-api-sandbox.circle.com/attestations/0xb9c36966bb8831928c5edc0ea929c4503403b8ff1aa8ab5e2c60a2890782cd54
3:29PM INF Broadcasting mint message from 0 to 4: with source tx hash 0x6882c80ebe162822e2b1720b76c303131f3d5ab3319381dd09d27d264734093f
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x103965fc0]

goroutine 66 [running]:
github.com/strangelove-ventures/noble-cctp-relayer/cmd/noble.Broadcast({{{{0x0, {...}, {...}, 0x3e8, 0x98d2ab, 0x64, 0x1}, {0x4, {...}, 0x3e8, ...}}, ...}, ...}, ...)
        /Users/justintieri/go/src/github.com/strangelove-ventures/noble-cctp-relayer/cmd/noble/broadcast.go:140 +0x900
github.com/strangelove-ventures/noble-cctp-relayer/cmd.StartProcessor({{{{0x0, {...}, {...}, 0x3e8, 0x98d2ab, 0x64, 0x1}, {0x4, {...}, 0x3e8, ...}}, ...}, ...}, ...)
        /Users/justintieri/go/src/github.com/strangelove-ventures/noble-cctp-relayer/cmd/process.go:147 +0x774
created by github.com/strangelove-ventures/noble-cctp-relayer/cmd.Start in goroutine 1
        /Users/justintieri/go/src/github.com/strangelove-ventures/noble-cctp-relayer/cmd/process.go:72 +0x1ec

```

--- a/cmd/ethereum/listener.go
+++ b/cmd/ethereum/listener.go
@@ -3,9 +3,12 @@ package ethereum
 import (
 	"bytes"
 	"context"
-	"cosmossdk.io/log"
 	"embed"
 	"fmt"
+	"math/big"
+	"os"
+
+	"cosmossdk.io/log"
 	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/common"
@@ -13,8 +16,6 @@ import (
 	"github.com/pascaldekloe/etherstream"
 	"github.com/strangelove-ventures/noble-cctp-relayer/config"
 	"github.com/strangelove-ventures/noble-cctp-relayer/types"
-	"math/big"
-	"os"
 )
 
 //go:embed abi/MessageTransmitter.json
@@ -66,7 +67,7 @@ func StartListener(cfg config.Config, logger log.Logger, processingQueue chan *t
 	for _, historicalLog := range history {
 		parsedMsg, err := types.EvmLogToMessageState(messageTransmitterABI, messageSent, &historicalLog)
 		if err != nil {
-			logger.Error("Unable to parse history log into MessageState, skipping")
+			logger.Error("Unable to parse history log into MessageState, skipping", "err", err)
 			continue
 		}
 		logger.Info(fmt.Sprintf("New historical msg from source domain %d with tx hash %s", parsedMsg.SourceDomain, parsedMsg.SourceTxHash))

--- a/cmd/noble/broadcast.go
+++ b/cmd/noble/broadcast.go
@@ -131,48 +131,72 @@ func Broadcast(
 		}
 
 		rpcResponse, err := rpcClient.BroadcastTxSync(context.Background(), txBytes)
-		if err == nil && rpcResponse.Code == 0 {
-			msg.Status = types.Complete
-			return rpcResponse, nil
-		}
+		if err != nil || (rpcResponse != nil && rpcResponse.Code != 0) {
+			// Log the error
+			logger.Error(fmt.Sprintf("error during broadcast: %s", getErrorString(err, rpcResponse)))
 
-		// check tx response code
-		logger.Error(fmt.Sprintf("received non zero : %d - %s", rpcResponse.Code, rpcResponse.Log))
-
-		if err == nil && rpcResponse.Code == 32 {
-			// on account sequence mismatch, extract correct account sequence and retry
-			pattern := `expected (\d+), got (\d+)`
-			re := regexp.MustCompile(pattern)
-			match := re.FindStringSubmatch(rpcResponse.Log)
-
-			var newAccountSequence int64
-			if len(match) == 3 {
-				// Extract the numbers from the match.
-				newAccountSequence, _ = strconv.ParseInt(match[1], 10, 0)
-			} else {
-				// otherwise, just request the account sequence
-				_, newAccountSequence, err = GetNobleAccountNumberSequence(cfg.Networks.Destination.Noble.API, nobleAddress)
-				if err != nil {
-					logger.Error("unable to retrieve account number")
-				}
+			if err != nil || rpcResponse == nil {
+				// Log retry information
+				logger.Info(fmt.Sprintf("Retrying in %d seconds", cfg.Networks.Destination.Noble.BroadcastRetryInterval))
+				time.Sleep(time.Duration(cfg.Networks.Destination.Noble.BroadcastRetryInterval) * time.Second)
+				continue
 			}
-			logger.Debug(fmt.Sprintf("error during broadcast: %s", rpcResponse.Log))
-			logger.Debug(fmt.Sprintf("retrying with new account sequence: %d", newAccountSequence))
-			sequenceMap.Put(cfg.Networks.Destination.Noble.DomainId, newAccountSequence)
-		}
-		if err != nil {
-			logger.Error(fmt.Sprintf("error during broadcast: %s", err.Error()))
+
+			// Log details for non-zero response code
+			logger.Error(fmt.Sprintf("received non-zero: %d - %s", rpcResponse.Code, rpcResponse.Log))
+
+			// Handle specific error code (32)
+			if rpcResponse.Code == 32 {
+				newAccountSequence := extractAccountSequence(logger, rpcResponse.Log, nobleAddress, cfg.Networks.Destination.Noble.API)
+				logger.Debug(fmt.Sprintf("retrying with new account sequence: %d", newAccountSequence))
+				sequenceMap.Put(cfg.Networks.Destination.Noble.DomainId, newAccountSequence)
+			}
+
+			// Log retry information
 			logger.Info(fmt.Sprintf("Retrying in %d seconds", cfg.Networks.Destination.Noble.BroadcastRetryInterval))
 			time.Sleep(time.Duration(cfg.Networks.Destination.Noble.BroadcastRetryInterval) * time.Second)
 			continue
 		}
 
-		logger.Info(fmt.Sprintf("Retrying in %d seconds", cfg.Networks.Destination.Noble.BroadcastRetryInterval))
-		time.Sleep(time.Duration(cfg.Networks.Destination.Noble.BroadcastRetryInterval) * time.Second)
+		// Tx was successfully broadcast
+		msg.Status = types.Complete
+		return rpcResponse, nil
 	}
+
 	msg.Status = types.Failed
 
 	return nil, errors.New("reached max number of broadcast attempts")
+}
+
+// getErrorString returns the appropriate value to log when tx broadcast errors are encountered.
+func getErrorString(err error, rpcResponse *ctypes.ResultBroadcastTx) string {
+	if rpcResponse != nil {
+		return rpcResponse.Log
+	}
+	return err.Error()
+}
+
+// extractAccountSequence attempts to extract the account sequence number from the RPC response logs when
+// account sequence mismatch errors are encountered. If the account sequence number cannot be extracted from the logs,
+// it is retrieved by making a request to the API endpoint.
+func extractAccountSequence(logger log.Logger, rpcResponseLog, nobleAddress, nobleAPI string) int64 {
+	pattern := `expected (\d+), got (\d+)`
+	re := regexp.MustCompile(pattern)
+	match := re.FindStringSubmatch(rpcResponseLog)
+
+	if len(match) == 3 {
+		// Extract the numbers from the match.
+		newAccountSequence, _ := strconv.ParseInt(match[1], 10, 0)
+		return newAccountSequence
+	}
+
+	// Otherwise, just request the account sequence
+	_, newAccountSequence, err := GetNobleAccountNumberSequence(nobleAPI, nobleAddress)
+	if err != nil {
+		logger.Error("unable to retrieve account number")
+	}
+
+	return newAccountSequence
 }
 
 // NewRPCClient initializes a new tendermint RPC client connected to the specified address.

--- a/cmd/noble/listener.go
+++ b/cmd/noble/listener.go
@@ -1,16 +1,17 @@
 package noble
 
 import (
-	"cosmossdk.io/log"
 	"encoding/json"
 	"fmt"
-	"github.com/strangelove-ventures/noble-cctp-relayer/config"
-	"github.com/strangelove-ventures/noble-cctp-relayer/types"
 	"io"
 	"net/http"
 	"strconv"
 	"sync"
 	"time"
+
+	"cosmossdk.io/log"
+	"github.com/strangelove-ventures/noble-cctp-relayer/config"
+	"github.com/strangelove-ventures/noble-cctp-relayer/types"
 )
 
 func StartListener(cfg config.Config, logger log.Logger, processingQueue chan *types.MessageState) {

--- a/go.mod
+++ b/go.mod
@@ -18,6 +18,7 @@ require (
 )
 
 require (
+	cosmossdk.io/math v1.1.2
 	github.com/circlefin/noble-cctp v0.0.0-20230911222715-829029fbba29
 	github.com/cometbft/cometbft v0.38.0
 	github.com/gin-gonic/gin v1.8.1
@@ -30,7 +31,6 @@ require (
 	cosmossdk.io/api v0.3.1 // indirect
 	cosmossdk.io/core v0.5.1 // indirect
 	cosmossdk.io/depinject v1.0.0-alpha.4 // indirect
-	cosmossdk.io/math v1.1.2 // indirect
 	filippo.io/edwards25519 v1.0.0 // indirect
 	github.com/4meepo/tagalign v1.3.2 // indirect
 	github.com/99designs/go-keychain v0.0.0-20191008050251-8e49817e8af4 // indirect

--- a/types/message_state.go
+++ b/types/message_state.go
@@ -6,13 +6,14 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"strconv"
+	"time"
+
 	"github.com/circlefin/noble-cctp/x/cctp/types"
 	"github.com/cosmos/cosmos-sdk/types/bech32"
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/crypto"
-	"strconv"
-	"time"
 )
 
 const (
@@ -45,7 +46,6 @@ type MessageState struct {
 
 // EvmLogToMessageState transforms an evm log into a messageState given an ABI
 func EvmLogToMessageState(abi abi.ABI, messageSent abi.Event, log *ethtypes.Log) (messageState *MessageState, err error) {
-
 	event := make(map[string]interface{})
 	_ = abi.UnpackIntoMap(event, messageSent.Name, log.Data)
 
@@ -80,7 +80,7 @@ func EvmLogToMessageState(abi abi.ABI, messageSent abi.Event, log *ethtypes.Log)
 		return messageState, nil
 	}
 
-	return nil, errors.New(fmt.Sprintf("unable to parse txn into message.  tx hash %s", log.TxHash.Hex()))
+	return nil, errors.New(fmt.Sprintf("unable to parse tx into message, tx hash %s", log.TxHash.Hex()))
 }
 
 // NobleLogToMessageState transforms a Noble log into a messageState

